bluesky/callbacks/best_effort.py

```diff
@@ -529,6 +529,46 @@ def stop(self, doc):
         self.check_visibility()
         super().stop(doc)
 
+class BaselinePrinter(Callback):
+    def __init__(self, start_doc, file=sys.stdout):
+        # We accept a start_doc but discard it.
+        ...
+        self._baseline_enabled = True
+        self._descriptors = {}
+        self._baseline_toggle = True
+        self._file = file
+    def descriptor(self, doc):
+        # TODO Check the stream name of this descriptor using doc['name']
+        # If the stream name is 'baseline', record doc['uid'] in an instance
+        # variable for future reference.
+        ...
+        self._descriptors[doc['uid']] = doc
```

Comment:
I think we can make `event` more efficient by checking the descriptor's name _once_ here instead of repeatedly every time we receive a new event. That is, move

```py
if descriptor.get('name') == 'baseline':
```

here and only stash the descriptor(s) if they have the name we are interested in. Just ignore the rest.